crates/oxc_sourcemap/src/encode.rs

```diff
@@ -125,14 +143,24 @@ fn encode_vlq(out: &mut String, num: i64) {
 fn test_encode() {
     let input = r#"{
         "version": 3,
-        "sources": ["coolstuff.js"],
+        "sources": ["coolstuff.js", "\0a.js"],
         "names": ["x","alert"],
         "mappings": "AAAA,GAAIA,GAAI,EACR,IAAIA,GAAK,EAAG,CACVC,MAAM"
     }"#;
     let sm = SourceMap::from_json_string(input).unwrap();
-    let sm2 = SourceMap::from_json_string(&sm.to_json_string()).unwrap();
+    let sm2 = SourceMap::from_json_string(&sm.to_json_string().unwrap()).unwrap();
 
     for (tok1, tok2) in sm.get_tokens().zip(sm2.get_tokens()) {
         assert_eq!(tok1, tok2);
     }
 }
+
+#[test]
+fn test_encode_escape_string() {
+    // '\0' should be escaped.
+    let sm = SourceMap::new(None, vec!["\0a.js".into()], vec![], None, vec![], None);
```

Comment:
You mentioned "Here using `serde_json::to_string` to serialization `names/source_contents/sources`."

I think the test case should cover all 3.